addons/zeusRecon/functions/fnc_zeusInterfaceOpenedEH.sqf

```diff
@@ -0,0 +1,43 @@
+#include "script_component.hpp"
+
+TRACE_1("zeusInterfaceOpenedEH",_this);
+
+if (!(missionNamespace getVariable [QGVAR(enabled), false])) exitWith {};
+
+if (isNil QGVAR(init)) then {
+    GVAR(state) = 0;
+    GVAR(stateSubIndex) = 0;
+
+    GVAR(uniqueID) = 0;
+    GVAR(allTargets) = [];
+    GVAR(allTargetInfo) = [];
+    GVAR(targetGroups) = [];
+    
+    GVAR(friendlyUnits) = [];
+    GVAR(friendlyGroups) = [];
+    GVAR(friendlyReports) = [];
+
+    GVAR(addToCurator) = [];
+    GVAR(removeFromCurator) = [];
+
+    if (isNull getAssignedCuratorLogic player) then {systemChat "Unknown Curator?";};
+
+    addMissionEventHandler ["EachFrame", {
```

Comment:
CBA state machine is designed for objects.
this loops through both objects and general arrays

Main advantage is we can control the update rate, right now it's 5x each frame (probably higher than it needs to be)